### ISSUE TITLE
BUG: Maintained check should verify who closed an issue

### ISSUE THREAD

**Issue by laurentsimon** [ossf/scorecard#1523]
I don't think https://github.com/ossf/scorecard/blob/main/checks/maintained.go#L93 is true. I think the creator of an issue can close it even if they are external to a repo and not a collaborator. I think we need to verify who closed the issue, if feasible.

**Comment by chrismcgehee** [comment]
Yeah, you're right; the creator can close their own issue. Now that I think about it a little more, maybe we could just remove the closed check. If a repo maintainer were to close an issue without giving any comment, I don't know that should be considered "meaningful" activity. If they were to comment and then close the issue, it would count as activity. So this is what our logic would be:

If a collaborator, owner, or member creates or comments on an issue in the last 90 days, count that as activity.

Wdut?

**Comment by azeemshaikh38** [comment]
+1 to removing the closed check for now. We can first see how the `AuthorAssociation` works using next week's cron job and revisit this if needed. @laurentsimon wdut?

**Comment by laurentsimon** [comment]
question: we've added fields like `AuthorAssociation *RepoAssociation` `Issue` and `IssueComment`.
Should these live under `User` instead?

It would be more natural to have:
```
// Issue represents a thread like GitHub issue comment thread.
type Issue struct {
	URI               *string
	CreatedAt         *time.Time
	Creator         User
	Comments          []IssueComment
}

// IssueComment represents a comment on an issue.
type IssueComment struct {
	CreatedAt         *time.Time
	Commenter User
}

type User struct{
   Login string
  AuthorAssociation *RepoAssociation
}
```

I don't know if the APIs lend themselves well to do that - I have not looked. I'd hope they do because author association should be associated to a user.
